File: src/lib/scala/Slate.Common/src/main/scala/slate/common/logging/LoggerConsole.scala
/**
<slate_header>
  author: Kishore Reddy
  url: https://github.com/kishorereddy/scala-slate
  copyright: 2015 Kishore Reddy
  license: https://github.com/kishorereddy/scala-slate/blob/master/LICENSE.md
  desc: a scala micro-framework
  usage: Please refer to license on github for more info.
</slate_header>
  */


package slate.common.logging

import slate.common.console.ConsoleWriter
import scala.reflect.runtime.universe.Type

class LoggerConsole(level:LogLevel = Debug,
                    name:String          = "console",
                    logType:Option[Type] = None )
  extends LoggerBase(level, name, logType) {

  private val _writer = new ConsoleWriter()


  /**
   * Logs to the console
 *
   * @param entry :
   */
  override protected def performLog(entry:LogEntry) =
  {
    val prefix = entry.
    level match {
      case Debug => _writer.subTitle  (entry.level.name + " : "  + entry.msg)
      case Info  => _writer.text      (entry.level.name + "  : " + entry.msg)
      case Warn  => _writer.url       (entry.level.name + "  : " + entry.msg)
      case Error => _writer.error     (entry.level.name + " : "  + entry.msg)
      case Fatal => _writer.highlight (entry.level.name + " : "  + entry.msg)
      case _     => _writer.text      (entry.level.name + " : "  + entry.msg)
    }
  }
}

File: src/lib/kotlin/slatekit-common/src/main/kotlin/slatekit/common/log/LoggerConsole.kt
/**
 * <slate_header>
 * url: www.slatekit.com
 * git: www.github.com/code-helix/slatekit
 * org: www.codehelix.co
 * author: Kishore Reddy
 * copyright: 2016 CodeHelix Solutions Inc.
 * license: refer to website and/or github
 * about: A tool-kit, utility library and server-backend
 * mantra: Simplicity above all else
 * </slate_header>
 */

package slatekit.common.log

import slatekit.common.DateTime

/**
 * Lightweight logger that prints to the console
 * using the ConsoleWriter in slate kit which has support
 * for colors and text semantics.
 * This is just used mostly for defaults.
 * You should be using the slatekit.providers module with support for logback
 */
class LoggerConsole(
    level: LogLevel = LogLevel.Debug,
    name: String = "console",
    logType: Class<*>? = null
) : Logger(level, name, logType) {

    /**
     * Logs to the console
     *
     * @param entry: 
     */
    override fun performLog(entry: LogEntry) {
        val prefix = "${entry.time} [$name] ${level.name}"
        when (entry.level) {
            LogLevel.Debug -> println("$prefix + : + ${entry.msg}")
            LogLevel.Info  -> println("$prefix + : + ${entry.msg}")
            LogLevel.Warn  -> println("$prefix + : + ${entry.msg}")
            LogLevel.Error -> println("$prefix + : + ${entry.msg}")
            LogLevel.Fatal -> println("$prefix + : + ${entry.msg}")
        }
    }
}
